spec/models/user_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should have_valid(:first_name).when("Jimmy", "Linc")}
+  it { should_not have_valid(:first_name).when(nil, "")}
+
+  it { should have_valid(:last_name).when("Ranger", "Kupke")}
+  it { should_not have_valid(:last_name).when(nil, "")}
+
+  it { should have_valid(:email).when("example@gmail.com", "erlich@aviato.net")}
+  it { should_not have_valid(:email).when("example", "example@", "exam@gma", "example.com", nil)}
+
+  describe "#admin?" do
+    context "when user is a admin" do
+      it "should give back true if user an admin" do
+        user = create(:user, role: "admin")
```

Comment:
I like seeing tests broken into two/ three stages with a single blank line in between. First section should be the setup (if any), next would be the action (if any), then the expectations. It's an organizational tool that helps your team/ future devs know what is going on in your test more explicitly. For a test this small, you could argue not to do it, but it's a great habit to start getting into!